packages/charts/react-charting/UnitTests/PlotlySchemaAdapterUT.test.tsx

```diff
@@ -0,0 +1,169 @@
+  });
+
+  test('transformPlotlyJsonToVSBCProps - Should return VSBC props', () => {
+    const plotlySchema = require('../src/components/DeclarativeChart/tests/schema/fluent_verticalstackedbarchart_test.json');
+    expect(transformPlotlyJsonToVSBCProps(plotlySchema, { current: colorMap }, true)).toMatchSnapshot();
+  });
+
+  test('transformPlotlyJsonToGVBCProps - Should return GVBC props', () => {
+    const plotlySchema = require('../src/components/DeclarativeChart/tests/schema/fluent_groupedverticalbarchart_test.json');
+    expect(transformPlotlyJsonToGVBCProps(plotlySchema, { current: colorMap }, true)).toMatchSnapshot();
+  });
+
+  test('transformPlotlyJsonToVBCProps - Should return VBC props', () => {
+    const plotlySchema = require('../src/components/DeclarativeChart/tests/schema/fluent_verticalbar_histogram_test.json');
+    expect(transformPlotlyJsonToVBCProps(plotlySchema, { current: colorMap }, true)).toMatchSnapshot();
+  });
+
+  test('transformPlotlyJsonToScatterChartProps - Should return line chart props', () => {
+    const plotlySchema = require('../src/components/DeclarativeChart/tests/schema/fluent_line_test.json');
+    expect(transformPlotlyJsonToScatterChartProps(plotlySchema, true, { current: colorMap }, true)).toMatchSnapshot();
+  });
+
+  test('transformPlotlyJsonToScatterChartProps - Should return area chart props', () => {
+    const plotlySchema = require('../src/components/DeclarativeChart/tests/schema/fluent_area_test.json');
+    expect(transformPlotlyJsonToScatterChartProps(plotlySchema, true, { current: colorMap }, true)).toMatchSnapshot();
+  });
+
+  test('transformPlotlyJsonToHorizontalBarWithAxisProps - Should return Horizontal bar chart with axis chart props', () => {
+    const plotlySchema = require('../src/components/DeclarativeChart/tests/schema/fluent_horizontalbar_test.json');
+    expect(
+      transformPlotlyJsonToHorizontalBarWithAxisProps(plotlySchema, { current: colorMap }, true),
+    ).toMatchSnapshot();
+  });
+
+  test('transformPlotlyJsonToHeatmapProps - Should return heatmap chart props', () => {
+    const plotlySchema = require('../src/components/DeclarativeChart/tests/schema/fluent_heatmap_test.json');
+    expect(transformPlotlyJsonToHeatmapProps(plotlySchema)).toMatchSnapshot();
+  });
+
+  test('transformPlotlyJsonToSankeyProps - Should return sankey chart props', () => {
+    const plotlySchema = require('../src/components/DeclarativeChart/tests/schema/fluent_sankey_test.json');
+    expect(transformPlotlyJsonToSankeyProps(plotlySchema, { current: colorMap }, true)).toMatchSnapshot();
+  });
+
+  test('transformPlotlyJsonToGaugeProps - Should return gauge chart props', () => {
+    const plotlySchema = require('../src/components/DeclarativeChart/tests/schema/fluent_gauge_test.json');
+    expect(transformPlotlyJsonToGaugeProps(plotlySchema, { current: colorMap }, true)).toMatchSnapshot();
+  });
+});
+
+runTest('sanitizeJson', () => {
```

Comment:
add test cases where the json is invalid.
Also add test case when json has 10 layers of nesting